### PR TITLE
Introduced Excel Reader

### PR DESCRIPTION
**Pull request by mandarmarathe123**
Reader to parse an Excel file into spark dataframe.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,3 +17,10 @@ Supported text formats are:
 * HTML Table
 
 Please see the detailed documentation [here](text/README.md).
+
+* Excel
+
+## excel
```
Comment (rahulbhatia023) [inline README.md:23]: The excel heading is coming twice. Should be present once only.

File: build.sbt [hunk 2]
```diff
@@ -80,6 +80,14 @@ val zioConfigDependencies = Seq(
...
 ).map(_ excludeAll ("org.scala-lang.modules", "scala-collection-compat"))
 
+val crealyticsDependencies = Seq(
+  "com.crealytics" %% "spark-excel" % "3.4.1_0.19.0"
+).map(_.cross(CrossVersion.for3Use2_13))
+
+val poiDependencies = Seq(
+  "org.apache.poi" % "poi" % "5.2.5"
+)
+
```
Comment (rahulbhatia023) [inline build.sbt:90]: The versions need to be mentioned under TOOL VERSIONS section in sbt file.

File: build.sbt [hunk 3]
```diff
@@ -97,9 +112,16 @@ lazy val `data-scalaxy-reader` = (project in file("."))
     publishLocal / skip := true
   )
   .aggregate(`reader-text`)
+  .aggregate(`reader-excel`)
```
Comment (rahulbhatia023) [inline build.sbt:115]: We can merge into single aggregate statement using **.aggregate(`reader-text`, `reader-excel`)**

File: build.sbt [hunk 4]
```diff
@@ -97,9 +112,16 @@ lazy val `data-scalaxy-reader` = (project in file("."))
...
   .settings(
     version := "2.0.0",
     libraryDependencies ++= textDependencies
   )
+
+lazy val `reader-excel` = (project in file("excel"))
+  .settings(
+    version := "1.0.0",
+    libraryDependencies ++= excelDependencies
+  )
```
Comment (rahulbhatia023) [inline build.sbt:127]: Add empty new line at the end of the file.

File: excel/src/test/scala/com/clairvoyant/data/scalaxy/reader/excel/ExcelToDataFrameReaderSpec.scala [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        fis.read(byteArray)
+        byteArray
+      }.get
+
+    val df = excelToDataFrameReader.read(
+      byteArray,
+      ExcelFormat(dataAddress = "'Transactions Report'!A2:G4")
+    )
+    df.count() shouldBe 2
+  }
```
Comment (rahulbhatia023) [inline excel/src/test/scala/com/clairvoyant/data/scalaxy/reader/excel/ExcelToDataFrameReaderSpec.scala:27]: add assertions to match actual and expected dataframe as well.

**Review by rahulbhatia023 — CHANGES_REQUESTED** [review]
Pls work on the review comments.

File: build.sbt (hunk 2 above)
Comment (mandarmarathe123) [reply build.sbt:92]: Added both Crealytics and PIO versions in separate section

File: README.md (hunk 1 above)
Comment (mandarmarathe123) [reply README.md:22]: Addressed.

File: excel/src/test/scala/com/clairvoyant/data/scalaxy/reader/excel/ExcelToDataFrameReaderSpec.scala [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.clairvoyant.data.scalaxy.test.util.matchers.DataFrameMatcher
+import com.clairvoyant.data.scalaxy.test.util.readers.DataFrameReader
+
+import java.io.FileInputStream
+import scala.util.Using
+
+class ExcelToDataFrameReaderSpec extends DataFrameReader with DataFrameMatcher {
+
+  val excelToDataFrameReader: ExcelToDataFrameReader.type = ExcelToDataFrameReader
```
[the remark this answers is not in the document]
Comment (mandarmarathe123) [reply excel/src/test/scala/com/clairvoyant/data/scalaxy/reader/excel/ExcelToDataFrameReaderSpec.scala]: Addressed.

**Comment by mandarmarathe123** [comment]
> Pls work on the review comments.

Done the necessary amendments and committed the code. Please have a look.

**Review by rahulbhatia023 — APPROVED** [review]

**Pull request merged**